render/vulkan/vulkan.c

```diff
@@ -163,21 +238,21 @@ struct wlr_vk_instance *vulkan_instance_create(size_t ext_count,
 	application_info.engineVersion = WLR_VERSION_NUM;
 	application_info.apiVersion = VK_API_VERSION_1_1;
 
-	const char *layers[] = {
-		"VK_LAYER_KHRONOS_validation",
-		// "VK_LAYER_RENDERDOC_Capture",
-		// "VK_LAYER_live_introspection",
-	};
-
-	unsigned layer_count = debug * (sizeof(layers) / sizeof(layers[0]));
+	uint32_t layer_count = 0;
+	char **layers = get_validation_layers(&layer_count);
+	wlr_log(WLR_DEBUG, "Using %"PRIu32" instance validation layers",
+		layer_count);
+	for (uint32_t i = 0; i < layer_count; ++i) {
+		wlr_log(WLR_DEBUG, "%s", layers[i]);
+	}
 
 	VkInstanceCreateInfo instance_info = {0};
 	instance_info.sType = VK_STRUCTURE_TYPE_INSTANCE_CREATE_INFO;
 	instance_info.pApplicationInfo = &application_info;
 	instance_info.enabledExtensionCount = ini->extension_count;
 	instance_info.ppEnabledExtensionNames = ini->extensions;
 	instance_info.enabledLayerCount = layer_count;
-	instance_info.ppEnabledLayerNames = layers;
+	instance_info.ppEnabledLayerNames = (const char *const *)layers;
```

Comment:
get_validation_layers should already return that type. We don't ever want to modify the layer names and it's needed if you want to use the string literals as suggested above.